prov/rxm/src/rxm_conn.c

```diff
@@ -318,15 +364,19 @@ static int
 rxm_conn_connect(struct util_ep *util_ep, struct util_cmap_handle *handle,
 		 const void *addr, size_t addrlen)
 {
-	struct rxm_ep *rxm_ep;
-	struct rxm_conn *rxm_conn;
 	struct fi_info *msg_info;
-	struct rxm_cm_data cm_data;
 	int ret;
-
-	rxm_ep = container_of(util_ep, struct rxm_ep, util_ep);
-
-	rxm_conn = container_of(handle, struct rxm_conn, handle);
+	struct rxm_ep *rxm_ep =
+		container_of(util_ep, struct rxm_ep, util_ep);
+	struct rxm_conn *rxm_conn =
+		container_of(handle, struct rxm_conn, handle);
+	struct rxm_cm_data cm_data = {
+		.proto = {
+			.ctrl_version = OFI_CTRL_VERSION,
+			.inject_size = rxm_ep->rxm_info->tx_attr->inject_size,
+			.sar_limit = rxm_ep->sar_limit,
```

Comment:
Do we need to exchange sar limit? If I understand correctly, the receiving side should handle individual segments and the last segment correctly without knowing the sar limit.

Reply:
You should probably at least exchange the size of the posted receive buffers.  That's needed to prevent buffer overruns.

Reply:
> You should probably at least exchange the size of the posted receive buffers. That's needed to prevent buffer overruns.

Yes, we need to exchange inject size (recv buffer size) but exchanging sar_limit seems unnecessary to me.
nit: can we rename inject_size to eager_size or eager_limit?

Reply:
> exchanging sar_limit seems unnecessary to me.

Makes sense to me

> can we rename inject_size to eager_size or eager_limit?

Sure